QuantConnect.TemplateBrokerage.Tests/TemplateFeatureRegressionAlgorithm.cs

```diff
@@ -555,5 +622,53 @@ protected virtual decimal GetOrderQuantity(Symbol symbol)
         {
             return Securities[symbol].SymbolProperties.MinimumOrderSize ?? 1;
         }
+
+        private bool SetContracts(Slice slice)
+        {
+            /// Check if the brokerage supports futures, if it does find a contract and add it to BrokerageAlgorithmSettings.FutureContract
+            if (BrokerageAlgorithmSettings.CanonicalFutureSymbol != null && !SetFutureContract(slice))
+            {
+                Debug($"{Time}: Waiting for future contract to be set...");
+                return false;
+            }
+
+            /// Check if the brokerage supports options, if it does find a contract and add it to BrokerageAlgorithmSettings.OptionContract
+            if (BrokerageAlgorithmSettings.CanonicalOptionSymbol != null && !SetOptionContract(BrokerageAlgorithmSettings.CanonicalOptionSymbol, slice.OptionChains.Values, SecurityType.Option, ref BrokerageAlgorithmSettings.OptionContract))
+            {
+                Debug($"{Time}: Waiting for option contract to be set...");
+                return false;
+            }
+
+            /// Check if the brokerage supports index option, if it does find a contract and add it to BrokerageAlgorithmSettings.IndexOptionContract
+            if (BrokerageAlgorithmSettings.CanonicalIndexOptionSymbol != null && !SetOptionContract(BrokerageAlgorithmSettings.CanonicalIndexOptionSymbol, slice.OptionChains.Values, SecurityType.IndexOption, ref BrokerageAlgorithmSettings.IndexOptionContract))
+            {
+                Debug($"{Time}: Waiting for index option contract to be set...");
+                return false;
+            }
+
+            /// Check if the brokerage supports future options, if it does find a contract and add it to BrokerageAlgorithmSettings.FutureOptionContract
+            if (BrokerageAlgorithmSettings.CanonicalFutureOptionSymbol != null && !SetOptionContract(BrokerageAlgorithmSettings.CanonicalFutureOptionSymbol, slice.OptionChains.Values, SecurityType.FutureOption, ref BrokerageAlgorithmSettings.FutureOptionContract))
+            {
+                Debug($"{Time}: Waiting for future option contract to be set...");
+                return false;
+            }
```

Comment:
Should this contain crypto futures as well?

Reply:
I thought the same, but searching in the regression algorithms, I noticed that you could make an order for a crypto future just with the symbol you have added; you didn't need a contract such as for options, futures, index options and future options.